src/main/java/org/opensearch/neuralsearch/search/collector/HybridTopFieldDocSortCollector.java

```diff
@@ -0,0 +1,403 @@
+    private FieldValueHitQueue.Entry bottom = null;
+    @Getter
+    private int totalHits;
+    protected int docBase;
+    protected LeafFieldComparator comparators[];
+    @Getter
+    @Setter
+    private TotalHits.Relation totalHitsRelation = TotalHits.Relation.EQUAL_TO;
+    /*
+      reverseMul is used to set the direction of the sorting when creating comparators.
+      In threshold check reverseMul is used in comparison logic.
+      It modifies the comparison of either reverse or maintain the natural order depending on its value.
+      This ensures that the compareBottom method adjusts the order based on whether you want ascending or descending sorting.
+     */
+    protected int reverseMul;
+    protected FieldValueHitQueue<FieldValueHitQueue.Entry>[] compoundScores;
+    protected boolean queueFull[];
+    @Getter
+    protected float maxScore = 0.0f;
+    protected int[] collectedHits;
+    /*
+       searchSortPartOfIndexSort is used to evaluate whether to perform index sort or not.
+     */
+    private Boolean searchSortPartOfIndexSort = null;
+
+    private static final TopFieldDocs EMPTY_TOPDOCS = new TopFieldDocs(
+        new TotalHits(0, TotalHits.Relation.EQUAL_TO),
+        new ScoreDoc[0],
+        new SortField[0]
+    );
+
+    // Declaring the constructor private prevents extending this class by anyone
+    // else. Note that the class cannot be final since it's extended by the
+    // internal versions. If someone will define a constructor with any other
+    // visibility, then anyone will be able to extend the class, which is not what
+    // we want.
+    HybridTopFieldDocSortCollector(final int numHits, final HitsThresholdChecker hitsThresholdChecker, final Sort sort) {
+        this.numHits = numHits;
+        this.hitsThresholdChecker = hitsThresholdChecker;
+        this.sort = sort;
+    }
+
+    /**
+     * HybridCollectorManager fetches the topDocs in the reduce method.
+     * @return List of TopFieldDocs which represents results of Top Docs of individual subquery.
+     */
+    public List<TopFieldDocs> topDocs() {
+        if (compoundScores == null) {
+            return new ArrayList<>();
+        }
+
+        List<TopFieldDocs> topFieldDocs = new ArrayList<>();
+        for (int subQueryNumber = 0; subQueryNumber < compoundScores.length; subQueryNumber++) {
+            topFieldDocs.add(
+                topDocsPerQuery(
+                    0,
+                    Math.min(collectedHits[subQueryNumber], compoundScores[subQueryNumber].size()),
+                    compoundScores[subQueryNumber],
+                    collectedHits[subQueryNumber],
+                    sort.getSort()
+                )
+            );
+        }
+        return topFieldDocs;
+    }
+
+    @Override
+    public ScoreMode scoreMode() {
+        return hitsThresholdChecker.scoreMode();
+    }
+
+    protected abstract class HybridTopDocSortLeafCollector implements LeafCollector {
```

Comment:
Why this class has to be part of this file? and why this class has to be abstract , when there is no one extending this class and there is no concrete implementation from this class.

Reply:
Because collect method which is used in SimpleFieldCollector and PagingFieldCollector is defined in LeafCollector interface.